File: bindings/node/examples/remote_chat.mjs
import {
  CogentClient,
  addOpenAiRemote,
  printText,
  readRemoteArgs,
  textOptions,
} from './_common.mjs';

const { model, input } = readRemoteArgs('Explain remote inference in one sentence.');
const client = new CogentClient();
const endpoint = addOpenAiRemote(client, model);
const run = client.chat({
  endpoint,
  messages: [
    { role: 'system', content: 'Answer concisely.' },
    { role: 'user', content: input },
  ],
  options: textOptions(),
  streamTokens: true,
});
let streamed = '';
for await (const batch of run.tokens) {
  process.stdout.write(batch.text);
  streamed += batch.text;
}
process.stdout.write('\n');
const result = await run.response;
if (streamed !== result.text) {
  throw new Error('streamed token batches did not match final response text');
}
printText(result);

File: bindings/python/examples/remote_chat.py
from __future__ import annotations

from cogentlm import ChatMessage

from _common import (
    CogentClient,
    add_openai_remote,
    print_text,
    read_remote_args,
    text_options,
)


def main() -> None:
    model, prompt = read_remote_args("Explain remote inference in one sentence.")
    client = CogentClient()
    endpoint = add_openai_remote(client, model)
    run = client.chat(
        [
            ChatMessage("system", "Answer concisely."),
            ChatMessage("user", prompt),
        ],
        endpoint=endpoint,
        options=text_options(),
        token_delivery="batch",
    )
    streamed = ""
    for batch in run.tokens():
        print(batch["text"], end="", flush=True)
        streamed += batch["text"]
    print()
    result = run.result()
    if streamed != result["text"]:
        raise RuntimeError("streamed token batches did not match final response text")
    print_text(result)


if __name__ == "__main__":
    main()
